util, refactor: Improve headers for bitcoin-wallet tool

hebasto opened the pull request:
This PR:
- removes unneeded `#include <wallet/wallet.h>` from `<wallet/wallettool.h>`
- introduces class forward declaration in `<wallet/wallettool.h>`
- added `#include <config/bitcoin-config.h>` to `wallet/wallettool.cpp` where the `USE_BDB` macro is used

hebasto committed:
util, refactor: Improve headers for bitcoin-wallet tool

stratospher reviewed (commented):
ACK 49b906c. Verified that:

- #include <wallet/wallet.h> isn’t needed in src/wallet/wallettool.h
- All the newly added #include in the files are necessary.

I’m curious to know why `WalletShowInfo()` was moved out of `src/wallet/wallettool.h` though.

hebasto commented:
Updated 49b906c842676a79c16d5def7588ac7aefb72ad8 -> 3431839c33fa3892c982322e4add39e28ddba719 ([pr23346.02](https://github.com/hebasto/bitcoin/commits/pr23346.02) -> [pr23346.03](https://github.com/hebasto/bitcoin/commits/pr23346.03),  [diff](https://github.com/hebasto/bitcoin/compare/pr23346.02..pr23346.03)):

- addressed @luke-jr's [comment](https://github.com/bitcoin/bitcoin/pull/23346#discussion_r739398954)

hebasto commented on src/bitcoin-wallet.cpp:
```diff
@@ -2,20 +2,23 @@
-#if defined(HAVE_CONFIG_H)
-#include <config/bitcoin-config.h>
-#endif
-
```
[Updated](https://github.com/bitcoin/bitcoin/pull/23346#issuecomment-955576224).

Sjors commented:
Lgtm, @achow101?

MarcoFalke commented on src/wallet/wallettool.cpp:
```diff
@@ -2,6 +2,12 @@
+#include <config/bitcoin-config.h>
+#endif
+
+#include <wallet/wallettool.h>
```
doesn't matter here, but the self-include is always the very first one on its own section for us, I think
